lib/cache_mngr.c

```diff
@@ -148,6 +148,33 @@ static int event_input(struct nl_msg *msg, void *arg)
  */
 int nl_cache_mngr_alloc(struct nl_sock *sk, int protocol, int flags,
 			struct nl_cache_mngr **result)
+{
+	struct nl_sock *sk_blocking = nl_socket_alloc();
+	int err;
+
+	if (!sk_blocking) {
+		return -NLE_NOMEM;
+	}
+	if ((err = nl_connect(sk_blocking, protocol)) < 0) {
+			nl_socket_free(sk_blocking);
+			return err;
+	}
+	return nl_cache_mngr_alloc_ex(sk, sk_blocking, protocol, flags, result);
+}
+
+/**
+ * Allocate new cache manager, with custom callback on refill socket
+ * @arg sk		Netlink socket or NULL to auto allocate
+ * @arg sk_blocking		Blocking Netlink socket for cache refills
+ * @arg protocol	Netlink protocol this manager is used for
+ * @arg flags		Flags (\c NL_AUTO_PROVIDE)
+ * @arg result		Result pointer
+ *
+ * Same as \f nl_cache_mngr_alloc, but sets custom refill socket
+ * Note: ownership of the sk_blocking passes to the cache manager
+ */
+int nl_cache_mngr_alloc_ex(struct nl_sock *sk, struct nl_sock *sk_blocking, int protocol, int flags,
```

Comment:
`sk_blocking` is internally assigned to `cm_sync_sock`. While it's unclear which is the best name, they probably should be consistent. I think the parameter should be called `sync_sk`.